examples/antarctic_slope_current_model.jl

```diff
@@ -0,0 +1,154 @@
+underwater_slope(x, y) = -1750tanh(0.00004*(y - 150000)) - 2250
+# TODO: add 50km troughs to the slope, dependent on x and y. Use appendix C to add detail here
+
+grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(underwater_slope))
+
+println(grid)
+
+# TODO: add underwater slope at y = 0 with troughs
+
+#
+# Creating the Hydrostatic model:
+#
+
+# Forcings:
+u_forcing(x, y, z, t) = exp(z) * cos(x) * sin(t) # Not actual forcing, just example
+
+# TODO: need to use forcings to enact the sponge layers. Their support is within 20000 m
+# of the N/S boundaries, they impose a cross-slope buoyancy gradient, and the relaxation
+# tiem scales decrease linearly with distance from the interior termination of the sponge layers
+
+# We'll use Relaxation() to impose a sponge layer forcing on velocity, temperature, and salinity
+damping_rate       = 1 / 43200 # Relaxation time scale in seconds, need to make this decrease linearly toward outermost boundary
+south_mask         = GaussianMask{:y}(center=0, width=sponge_width)
+north_mask         = GaussianMask{:y}(center=Ly, width=sponge_width)
+south_sponge_layer = Relaxation(; rate=damping_rate, mask=south_mask)
+north_sponge_layer = Relaxation(; rate=damping_rate, mask=north_mask)
+sponge_layers      = (south_sponge_layer, north_sponge_layer)
+# TODO: compose north_mask and south_mask together into one sponge layer, OR compose north/south sponge layers
+
+
+# Boundary Conditions:
+no_slip_bc   = ValueBoundaryCondition(0.0)
+free_slip_bc = FluxBoundaryCondition(nothing)
+
+free_slip_surface_bcs = FieldBoundaryConditions(no_slip_bc, top=FluxBoundaryCondition(nothing))
+no_slip_field_bcs     = FieldBoundaryConditions(no_slip_bc)
+
+# Buoyancy Equations of State - we want high order polynomials, so we'll use TEOS-10
+eos = TEOS10EquationOfState()
+
+# Coriolis Effect, using basic f-plane with precribed reference Coriolis parameter
+coriolis = FPlane(f=-1.3e14)
+
+# Diffusivities as part of closure
+# TODO: make sure this works for biharmonic diffusivities as the horizontal, 
+horizontal_closure = HorizontalScalarDiffusivity(ν=0.1, κ=0.1)
+vertical_closure   = VerticalScalarDiffusivity(ν=3e-4, κ=1e-5)
+
+# Assuming no particles or biogeochemistry
+model = HydrostaticFreeSurfaceModel(; grid,
+                                         clock = Clock{eltype(grid)}(0, 0, 1),
+                            momentum_advection = CenteredSecondOrder(),
+                              tracer_advection = CenteredSecondOrder(),
+                                      buoyancy = SeawaterBuoyancy(equation_of_state=eos),
+                                      coriolis = coriolis,
+                                  free_surface = ImplicitFreeSurface(gravitational_acceleration=g_Earth),
+                                       forcing = (u=sponge_layers, v=sponge_layers, w=sponge_layers, T=sponge_layers, S=sponge_layers), # NamedTuple()
+                                       closure = (horizontal_closure, vertical_closure),
```

Comment:
We can rely entirely on WENO for horizontal dissipation. The only closure we need is for vertical mixing.